### PR TITLE
FALCON-1825 Process end time not exclusive in case of Native Scheduler

### PR DESCRIPTION
**Pull request by pavankumar526**

### CODE REVIEW

File: scheduler/src/main/java/org/apache/falcon/notification/service/impl/AlarmService.java [hunk 1]
```diff
@@ -87,7 +87,7 @@ public void register(NotificationRequest notificationRequest) throws Notificatio
         DateTime nextStartTime = request.getStartTime();
         DateTime endTime;
         if (request.getEndTime().isBefore(currentTime)) {
-            endTime = request.getEndTime();
+            endTime = request.getEndTime().minusMinutes(1);
```
Comment (pallavi-rao) [inline scheduler/src/main/java/org/apache/falcon/notification/service/impl/AlarmService.java]: Should we minus the minute after if-else? Also, this change is causing some UT failures, please fix those.

Comment (pavankumar526) [inline scheduler/src/main/java/org/apache/falcon/notification/service/impl/AlarmService.java]: No we shouldn't, We can run current time instance if time is current.

Comment (pavankumar526) [inline scheduler/src/main/java/org/apache/falcon/notification/service/impl/AlarmService.java]: I will fix test case and update patch.

File: scheduler/src/main/java/org/apache/falcon/notification/service/impl/AlarmService.java [hunk 2]
```diff
@@ -109,7 +109,7 @@ public void register(NotificationRequest notificationRequest) throws Notificatio
             }
         }
         // All past events have been scheduled. Nothing to schedule in the future.
-        if (request.getEndTime().isBefore(nextStartTime)) {
+        if (endTime.isBefore(nextStartTime)) {
```
Comment (pallavi-rao) [inline scheduler/src/main/java/org/apache/falcon/notification/service/impl/AlarmService.java]: request.getEndTime is also being used in creating trigger for the quartz job. Doesn't that need to be changed or is it already exclusive in quartz?

Comment (pavankumar526) [inline scheduler/src/main/java/org/apache/falcon/notification/service/impl/AlarmService.java]: It is exclusive in quartz, that why not added.

Comment (pallavi-rao) [inline scheduler/src/main/java/org/apache/falcon/notification/service/impl/AlarmService.java]: ok. +1 for the patch then.

Comment (pavankumar526) [inline scheduler/src/main/java/org/apache/falcon/notification/service/impl/AlarmService.java]: Will merge shortly

**Pull request closed without merging**